src/runtime_src/core/include/xclerr.h

```diff
@@ -88,4 +92,22 @@ struct xclErrorStatus {
         unsigned mFirewallLevel;
 };
 
+/**
+ * struct xclErrorLast - Container for all last(latest) error records
+ * Only one entry in error array per error class xrtErrorClass
+ * A xrtErrorModule may produce multiple classes of errors
+ * xrtErrorCode (64 bits) = ErrorNum + Driver + Severity + Module + Class
+ */
+struct xclErrorLast {
+	xrtErrorCode	err_code;	/* 64 bits; XRT error code */
+	xrtErrorTime	ts;		/* 64 bits; timestamp */
+	unsigned        pid;            /* 32 bits; pid associated with error, if available */
+};
+
+struct xocl_errors {
+	int		num_err;	/* number of errors recorded */
+	int		cap;		/* capacity of current error array */
+	struct xclErrorLast *errors;	/* error array pointer */
+};
+
```

Comment:
What is the intended end-use of these structs?  I am concerned that these structs are added to an exported public header file and that we will have application code dependencies on struct and their sizes.

Please provide the bigger picture here in terms of how these structs will be used and by whom.

Reply:
updated based on discussion